plugins/reports/monographReport/MonographReportPlugin.inc.php

```diff
@@ -0,0 +1,177 @@
+	 */
+	public function getName(): string
+	{
+		return static::class;
+	}
+
+	/**
+	 * @copydoc Plugin::getDisplayName()
+	 */
+	public function getDisplayName(): string
+	{
+		return __('plugins.reports.monographReport.displayName');
+	}
+
+	/**
+	 * @copydoc Plugin::getDescription()
+	 */
+	public function getDescription(): string
+	{
+		return __('plugins.reports.monographReport.description');
+	}
+
+	/**
+	 * @copydoc ReportPlugin::display()
+	 */
+	public function display(array $args, Request $request): void
+	{
+		$press = $request->getContext();
+		if (!$press) {
+			throw new Exception('The monograph report requires a context');
+		}
+
+		header('content-type: text/comma-separated-values');
+		header('content-disposition: attachment; filename=monograph-report-' . date('Ymd') . '.csv');
+
+		$fields = $this->_getFieldMapper($press, $request);
+		$output = new SplFileObject('php://output', 'w');
+		try {
+			// UTF-8 BOM to force the file to be read with the right encoding
+			$output->fwrite("\xEF\xBB\xBF");
+			$output->fputcsv(array_keys($fields));
+
+			import('classes.submission.Submission'); // STATUS_PUBLISHED constant
+			$monographs = Services::get('submission')->getMany([
+				'contextId' => $press->getId(),
+				'status' => STATUS_PUBLISHED,
+				'orderBy' => 'dateSubmitted',
+				'orderDirection' => 'ASC'
+			]);
+			/** @var Submission */
+			foreach ($monographs as $monograph) {
+				$row = [];
+				foreach ($fields as $getter) {
+					$row[] = $getter($monograph->getCurrentPublication());
+				}
+				$output->fputcsv($row);
+			}
+		} finally {
+			$output = null;
+		}
+	}
+
+	/**
+	 * Retrieves a dictionary, where the keys represent the column titles and the values a value getter
+	 */
+	private function _getFieldMapper(Press $press, Request $request): array
+	{
+		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_SUBMISSION);
+
+		/** @var SeriesDAO */
+		$seriesDAO = DAORegistry::getDAO('SeriesDAO');
+		/** @var CategoryDAO */
+		$categoryDao = DAORegistry::getDAO('CategoryDAO');
+
+		/** @var Series[] */
+		$seriesList = $seriesDAO->getByContextId($press->getId())->toAssociativeArray();
+		/** @var Category[] */
+		$categoryList = $categoryDao->getByContextId($press->getId())->toAssociativeArray();
+
+		return [
+			__('common.publication') => function (Publication $publication) {
+				return $publication->getLocalizedTitle();
+			},
+			__('submission.authors') => function (Publication $publication) {
+				$authors = [];
+				/** @var Author */
+				foreach ($publication->getData('authors') ?? [] as $author) {
+					$authors[] = $author->getFullName();
+				}
+				return implode("\n", $authors);
+			},
+			__('catalog.published') => function (Publication $publication) {
+				return $publication->getData('datePublished');
+			},
+			__('metadata.property.displayName.series') => function (Publication $publication) use ($seriesList) {
+				return ($series = $seriesList[$publication->getData('seriesId')] ?? null)
```

Comment:
What if there is no such index (`$publication->getData('seriesId')`) of the array `seriesList`?

Reply:
Ah, this is fine, because:
`echo $nullVariable['key'];` > Notice
`echo $nullVariable['key'] ?? 'Ok';` > Ok

It just doesn't work well with method calls:
`$nullVariable->method() ?? 'default'` > Exception